### DISCUSSION TITLE
libdisplaydevice is a WIP library - How to create a tool with this library (EnumAvailableDevices)

### DISCUSSION THREAD

**Discussion by comsyspro** [LizardByte/community#530] [Libraries]
### Select Topic Area

Question

### Body

How could I create a tool (exe) for Windows which will output the same as this TEST_F_S(EnumAvailableDevices) function here https://github.com/LizardByte/libdisplaydevice/blob/48ca85e8e7c3c01fef5db7f0befced7ea21d6238/tests/unit/windows/test_win_playground.cpp#L49
This code creates a gtest exe but I want to do it with a normal exe which runs the function in a int main(int argc, char** argv) { } function.

**Comment by ReenigneArcher** [comment]
This is a library that is meant to be integrated into another application. Teaching how to do that from the ground would be a huge undertaking (without knowing your skill level or anything).

What I can do is point you to the implementation PR for Sunshine: https://github.com/LizardByte/Sunshine/pull/2894

And the docs: https://docs.lizardbyte.dev/projects/libdisplaydevice/master/

**Comment by comsyspro** [comment]
I think I could get started with it if there were an example how to include the library and how to execute a function like EnumAvailableDevices. At the moment I can read the docs but without such an example it would be hard to begin with. On the other hand I also could use the test_libdisplaydevice.exe which also outputs results but also adds some lines of gtest debug infos so the output needs to be cleaned parsing it to get a proper json string. Could you implement an example code or do you already have an example source code in your projects? Thank you for more infos about that.

**Comment by comsyspro** [comment]
I got something out of it now and could solve it. I also had to find out how to set the logging level so the output now prints clean json.

**Reply by ReenigneArcher** [reply]
Glad to hear! This is still a very new library (so far has never really been implemented into anything). I'm sure better documentation will come once it's more stable. If you'd like to contribute to the docs, or a basic example program, a PR would be welcomed.
